phpBB/phpbb/db/driver/driver.php

```diff
@@ -793,6 +804,98 @@ function sql_build_query($query, $array)
 		return $sql;
 	}
 
+
+	protected function _process_boolean_tree_first($operations_ary)
+	{
+		if ($operations_ary[0] !== 'AND' &&
+			$operations_ary[0] !== 'OR')
+		{
+			$operations_ary = array('AND', $operations_ary);
```

Comment:
throw an error, if it something that is unknown
missing NOT from the allowed list.